persistent/tests/test_persistence.py

```diff
@@ -166,6 +228,14 @@ def _test():
             inst._p_oid = new_OID
         self.assertRaises(ValueError, _test)
 
+    def test_assign_p_oid_not_in_cache_allowed(self):
+        jar = self._makeJar()
+        inst = self._makeOne()
+        inst._p_jar = jar
+        inst._p_oid = 1 # anything goes
+        inst._p_oid = 42
+        self.assertEqual(inst._p_oid, 42)
```

Comment:
I thought @strichter provided a very convincing use-case for allowing arbitrary values in `_p_oid` in his earlier comment.